src/Telemetry/Telemetry.cs

```diff
@@ -21,223 +22,81 @@ public static class Telemetry
-            /// </summary>
-            public const string DeviceAuthentication = "device_authentication";
-
-            /// <summary>
-            /// device_authentication_failure
-            /// </summary>
-            public const string DeviceAuthenticationFailure = "device_authentication_failure";
-
-            /// <summary>
-            /// dynamic_identityprovider_validation
-            /// </summary>
-            public const string DynamicIdentityProviderValidation = "dynamic_identityprovider_validation";
-
-            /// <summary>
-            /// dynamic_identityprovider_validation_failure
-            /// </summary>
-            public const string DynamicIdentityProviderValidationFailure = "dynamic_identityprovider_validation_failure";
-
-            /// <summary>
-            /// introspection
-            /// </summary>
-            public const string Introspection = "introspection";
-
-            /// <summary>
-            /// introspection_failure
-            /// </summary>
-            public const string IntrospectionFailure = "introspection_failure";
-
-            /// <summary>
-            /// pushed_authorization_request
-            /// </summary>
-            public const string PushedAuthorizationRequest = "pushed_authorization_request";
-
-            /// <summary>
-            /// pushed_authorization_request_failure
-            /// </summary>
-            public const string PushedAuthorizationRequestFailure = "pushed_authorization_request_failure";
-
-            /// <summary>
-            /// resourceowner_authentication
-            /// </summary>
-            public const string ResourceOwnerAuthentication = "resourceowner_authentication";
-
-            /// <summary>
-            /// resourceowner_authentication_failure
-            /// </summary>
-            public const string ResourceOwnerAuthenticationFailure = "resourceowner_authentication_failure";
-
-            /// <summary>
-            /// revocation
-            /// </summary>
-            public const string Revocation = "revocation";
-
-            /// <summary>
-            /// revocation_failure
-            /// </summary>
-            public const string RevocationFailure = "revocation_failure";
-
-            /// <summary>
-            /// token_issued
-            /// </summary>
-            public const string TokenIssued = "token_issued";
-
-            /// <summary>
-            /// token_issued_failure
-            /// </summary>
-            public const string TokenIssuedFailure = "token_issued_failure";
-
-            /// <summary>
-            /// unhandled_exception
-            /// </summary>
-            public const string UnHandledException = "unhandled_exception";
+            public const string Operation = "tokenservice.operation";
+            public const string ActiveRequests = "tokenservice.active_requests";
+            public const string ApiSecretValidation = "tokenservice.api.secret_validation";
```

Comment:
I'm not sure, that I understand the logic between . and _ in some of the names:
Assuming . is similar to a namespace, then it should be both .
"tokenservice.device_authentication" has an _
"tokenservice.resourceowner.authentication" has a .

Reply:
@Tornhoof The `.` is a namespace separator, the `_` is a word separator. Some tools creates a drill-down menu based on the dots. `device_authentication` and `resourceowner.authentication` should probably be the same, thanks for noticing.